pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -354,8 +354,12 @@ private CompletableFuture<Void> revokePermissionsAsync(String topicUri, String r
                             || !policies.auth_policies.getTopicAuthentication().get(topicUri).containsKey(role)) {
                         log.warn("[{}] Failed to revoke permission from role {} on topic: Not set at topic level {}",
                                 clientAppId(), role, topicUri);
-                        return FutureUtil.failedFuture(new RestException(Status.PRECONDITION_FAILED,
-                                "Permissions are not set at the topic level"));
+                        if (force) {
```

Comment:
Hi @michaeljmarshall and @TakaHiR07, I don't suggest throwing an exception when the policy doesn't exist.

What do you think of this?

Reply:
Sure, it's worth discussing if we should fail a `DELETE` request for something that is already deleted. Do we have a common pattern in Pulsar already for this design choice? My preference is to be consistent with the rest of the API.

Reply:
Similar to https://github.com/apache/pulsar/blob/f1d11586e96344c40cab923f0ed89961d673cd5c/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java#L4963-L4973

Reply:
I prefer to keep the same behavior as pulsar old version temporarily. We can enhance this exception throw in the refactor of permission module